Packs/ArmisEventCollector/Integrations/ArmisEventCollector/ArmisEventCollector.py

```diff
@@ -0,0 +1,427 @@
+import demistomock as demisto
+from CommonServerPython import *
+import urllib3
+from typing import Any
+
+# Disable insecure warnings
+urllib3.disable_warnings()
+
+
+''' CONSTANTS '''
+
+
+DATE_FORMAT = '%Y-%m-%dT%H:%M:%S'
+VENDOR = 'armis'
+PRODUCT = 'security'
+API_V1_ENDPOINT = '/api/v1'
+DEFAULT_MAX_FETCH = 1000
+EVENT_TYPES = {
+    'Alerts': {
+        'unique_id_key': 'alertId',
+        'aql_query': 'in:alerts',
+        'type': 'alerts'},
+    'Threat activities': {
+        'unique_id_key': 'activityUUID',
+        'aql_query': 'in:activity type:"Threat Detected"',
+        'type': 'threat_activities'}}
+
+
+''' CLIENT CLASS '''
+
+
+class Client(BaseClient):
+    """Client class to interact with Armis API - this Client implements API calls"""
+
+    def __init__(self, base_url, api_key, access_token, verify=False, proxy=False):
+        self._api_key = api_key
+        super().__init__(base_url=base_url, verify=verify, proxy=proxy)
+        if not access_token or not self.is_valid_access_token(access_token):
+            access_token = self.get_access_token()
+        self.update_access_token(access_token)
+
+    def update_access_token(self, access_token=None):
+        if not access_token:
+            access_token = self.get_access_token()
+        headers = {
+            'Authorization': f'{access_token}',
+            "Accept": "application/json"
+        }
+        self._headers = headers
+        self._access_token = access_token
+
+    def fetch_by_aql_query(self, aql_query: str, max_fetch: int, after: None | datetime = None):
+        """ Fetches events using AQL query.
+
+        Args:
+            aql_query (str): AQL query request parameter for the API call.
+            max_fetch (int): Max number of events to fetch.
+            after (None | datetime): The date and time to fetch events from. Defaults to None.
+
+        Returns:
+            list[dict]: List of events objects represented as dictionaries.
+        """
+        params: dict[str, Any] = {'aql': aql_query, 'includeTotal': 'true', 'length': max_fetch, 'orderBy': 'time'}
+        if after:  # if there is a time frame thats relative to last run
+            params['aql'] += f' after:{after.strftime(DATE_FORMAT)}'
+
+        # make first request to get first page of threat activity
+        raw_response = self._http_request(url_suffix='/search/', method='GET', params=params, headers=self._headers)
+        results = raw_response.get('data', {}).get('results', [])
```

Comment:
what if I asked for 50 events but first page has 1000?

Reply:
The page size is not static and is determined by the max_fetch paraemter, therefore, the mentioned scenario should not occur.
For example, if the max_fetch parameter is set to 50, the page size will be limited to a maximum of 50 for each request (or potentially fewer if no more events are available). However, if the max_fetch parameter exceeds 5,000 (which is the maximum page size limit of the API), the pagination mechanism will start working and handle paginataion untill the desierd number of max_events is met.